In [2]:
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import numpy as np
import scipy.stats as stats

## 1.1A
Datasets are 4 text files stored in csv format. Each record is on single line and attributes are separated with TAB character. All 4 datasets contain common attribute imei which makes relation within tables.

### Connections csv

In [5]:
connections = pd.read_csv('data/connections.csv', sep='\t')
#connections = sns.load_dataset("data/connections.csv")

connections.head()


ts                 imei  mwra  c.android.youtube  \
0  2018-05-05 10:00:00  3590433799317661107   0.0           11.65403   
1  2018-05-05 10:01:00  3590433799317662063   1.0           10.29551   
2  2018-05-05 10:02:00   863033069630348313   0.0           12.12831   
3  2018-05-05 10:03:00   359043379931766114   1.0           11.06419   
4  2018-05-05 10:04:00  3590433799317661842   1.0           12.08457   

   c.dogalize  c.android.gm  c.katana  c.android.chrome  c.raider  \
0    10.65335      10.48791  16.91620           5.04564  36.21508   
1    10.46363      10.98438  15.65637          14.82931  24.83765   
2     9.53752       6.78080   6.98948          10.92433   3.29441   
3    11.55759       8.87744  11.84499          15.45902  24.78878   
4    10.42860       9.82241  12.48869          12.85363  40.98237   

   c.android.vending  c.UCMobile.intl  c.UCMobile.x86  c.updateassist  
0           14.24956          4.09319        15.20934         5.11764  
1           57.49911         23.94919         8.82448        44.13570  
2           20.37891         52.55353        49.51037        99.75215  
3           36.40299         25.77644        21.84167        65.17774  
4           74.13316         96.77450        18.48116        90.77304

In [6]:
connections.shape

(15074, 13)

Connections csv has 15074 records (rows) and 13 attributes (columns). 

In [7]:
connections.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15074 entries, 0 to 15073
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   ts                 15074 non-null  object 
 1   imei               15074 non-null  int64  
 2   mwra               15074 non-null  float64
 3   c.android.youtube  15074 non-null  float64
 4   c.dogalize         15074 non-null  float64
 5   c.android.gm       15074 non-null  float64
 6   c.katana           15074 non-null  float64
 7   c.android.chrome   15074 non-null  float64
 8   c.raider           15074 non-null  float64
 9   c.android.vending  15074 non-null  float64
 10  c.UCMobile.intl    15074 non-null  float64
 11  c.UCMobile.x86     15074 non-null  float64
 12  c.updateassist     15074 non-null  float64
dtypes: float64(11), int64(1), object(1)
memory usage: 1.5+ MB


This data has no null values (the number of Non-Null values for each column is the same as the total number of rows). First column contains values of type object (which represents timestamp) and  will be possibly converted to type timestamp during further processing. The second attribute contains values of type int64 and stores International Mobile Equipment Identity number. The third one contains values of type float64, however obtains only one of two values: 1. or 0. representing malware-related-activity, so it will be probably converted to type boolean. Other 10 columns contain float64 values, which represent input value for evaluation of mwra. First three attributes (ts, imei, mwra) are discrete (categorical) attributes and the rest are continuous (numeric) values. 

The pair of first and second column (ts, imei) are keys for snapshot of the rest of values. The third column is the result of evaluation. 

In [ ]:
# oprava dat 1.2:
#connections['ts'] = pd.to_datetime(connections['ts'])
# maybe also mwra to boolean

### Devices csv

In [8]:
devices = pd.read_csv('data/devices.csv', sep='\t')
devices.head()

latitude  longitude          store_name code  \
0  18.57677  -72.22625  Croix-des-Bouquets   HT   
1 -25.54073  152.70493         Maryborough   AU   
2  50.88506    4.07601         Denderleeuw   BE   
3  48.15659   28.28489              Soroca   MD   
4 -34.64966  -58.38341            Barracas   AR   

                         location                 imei  
0          America/Port-au-Prince  3590433799317662113  
1              Australia/Brisbane  8630330696303481107  
2                 Europe/Brussels  8630330696303481461  
3                 Europe/Chisinau  3590433799317661016  
4  America/Argentina/Buenos_Aires  3590433799317662253

In [9]:
devices.shape

(2895, 6)

Devices csv has 2895 records (rows) and 6 attributes (columns). 

In [10]:
devices.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2895 entries, 0 to 2894
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   latitude    2895 non-null   float64
 1   longitude   2895 non-null   float64
 2   store_name  2895 non-null   object 
 3   code        2892 non-null   object 
 4   location    2895 non-null   object 
 5   imei        2895 non-null   int64  
dtypes: float64(2), int64(1), object(3)
memory usage: 135.8+ KB


This data has 3 null values in attribute code. First and second attribute contains geolocation latitude and longitude position. Their values are float64. Next three attributes contain values of type object. They are strings which represent identification of store which sold the device. Attribute store_name contains name of the store, code is the code of the country, in which the store is located and the location is the name of the continent and the city in which the device was sold. In further processing, the location attribute might be split to two columns. The last attribute contains values of type int64 and stores International Mobile Equipment Identity number. All attributes are discrete (categorical). 

The attribute imei is the key for this table.

### Processes csv

In [ ]:
processes = pd.read_csv('data/processes.csv', sep='\t')
processes.shape

In [ ]:
processes.head()

### Profiles csv

In [ ]:
profiles = pd.read_csv('data/profiles.csv', sep='\t')
profiles.shape

In [ ]:
profiles.head()